std/math/bits/naf.go

```diff
@@ -0,0 +1,122 @@
+package bits
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark/backend/hint"
+	"github.com/consensys/gnark/frontend"
+)
+
+var NNAF = hint.NewStaticHint(nNaf)
+
+func init() {
+	hint.Register(NNAF)
+}
+
+// ToNAF retuns the naf decomposition of given input
```

Comment:
NAF decomposition needs a bit explanation (when I google, then there are no obvious answers on the first page at least). In NAF form, the input is decomposed into -1, 0, 1?